CRM/Utils/Type.php

```diff
@@ -435,12 +437,13 @@ public static function validate($data, $type, $abort = TRUE, $name = 'One of par
         break;
 
       case 'ContactReference':
+      case 'EntityReference':
         // null is valid
         if (strlen(trim($data)) == 0) {
           return trim($data);
         }
 
-        if (CRM_Utils_Rule::validContact($data)) {
+        if (CRM_Utils_Rule::positiveInteger($data)) {
```

Comment:
The function `validContact()` sounds like a contact-specific function, but all it does is call `positiveInteger` after some unnecessary parameter juggling.